src/rateLimiter.ts

```diff
@@ -29,19 +29,20 @@ import { RedisStore } from "./redisStore.ts";
 export const limit = <C extends Context, RT extends RedisType>(
   userOptions?: OptionsInterface<C, RT>,
 ) => {
-  const options = { ...defaultOptions, ...userOptions };
+  const options = { ...defaultOptions, ...(userOptions ?? {}) };
```

Comment:
```suggestion
  const options = { ...defaultOptions, ...userOptions };
```

There is no need to do an extra check + object creation here. If the `userOptions` is null or undefined, no errors will be thrown by the runtime. It's already valid syntax.